features/FEATURE_CLIENT/mbed-client-classic/source/m2mconnectionhandlerpimpl.cpp

```diff
@@ -310,9 +312,7 @@ int8_t M2MConnectionHandlerPimpl::connection_tasklet_handler()
 
 void M2MConnectionHandlerPimpl::socket_event()
```

Comment:
I haven't measured their performance, it would be interesting to see the numbers.

Although the issue isn't with average performance, but with jitter. As a simple linked-list, the memory pool garuntees constant runtime, where malloc may have an outlier (in the case of coalescing lazily for example) but be faster on average. This outlier may disrupt interrupt based communication.